digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadwayChangesDAO.scala

```diff
@@ -540,24 +540,24 @@ SELECT
 
       val elyCondition = {
         if (ely.nonEmpty)
-          s" AND rc.new_ely = ${ely.get}"
+          s" AND (rc.new_ely = ${ely.get} OR rc.old_ely = ${ely.get})"
         else
           ""
       }
 
       val roadNumberCondition = {
         if (roadNumber.nonEmpty)
-          s" AND rc.new_road_number = ${roadNumber.get}"
+          s" AND (rc.new_road_number = ${roadNumber.get} OR rc.old_road_number = ${roadNumber.get})"
         else
           ""
       }
 
       val roadPartCondition = {
         val parts = (minRoadPartNumber, maxRoadPartNumber)
         parts match {
-          case (Some(minPart), Some(maxPart)) => s"AND rc.new_road_part_number BETWEEN $minPart AND $maxPart"
-          case (None, Some(maxPart)) => s"AND rc.new_road_part_number <= $maxPart"
-          case (Some(minPart), None) => s"AND rc.new_road_part_number >= $minPart"
+          case (Some(minPart), Some(maxPart)) => s"AND (rc.new_road_part_number BETWEEN $minPart AND $maxPart OR rc.old_road_part_number BETWEEN $minPart AND $maxPart)"
+          case (None, Some(maxPart)) => s"AND (rc.new_road_part_number <= $maxPart OR rc.old_road_part_number <= $maxPart)"
+          case (Some(minPart), None) => s"AND (rc.new_road_part_number >= $minPart OR rc.old_road_part_number <= $minPart)"
```

Comment:
"<=" toisinpäin, copy-paste-härö?

Reply:
jos vain maxPart annettu, haetaan sitä pienemmät tai yht.suur eli <= max part
jos vain minPart annettu, haetaan sitä suuremmat tai yht.suur eli >= minPart

Reply:
Ei siis copy-paste-härö vaan ihan tarkoituksellinen

Reply:
Siis tämä:

rivi 560:
"OR rc.old_road_part_number <= $minPart"

Reply:
Oho silmät olivat jo lomalla näköjään 😀